core/domain/search_services.py

```diff
@@ -0,0 +1,201 @@
+"""Commands that can be used to manipulate search indices.
+
+"""
+from core.platform import models
+from core.domain import rights_manager
+
+search_services = models.Registry.import_search_services()
+(exp_models, feedback_models, user_models) = models.Registry.import_models([
+    models.NAMES.exploration, models.NAMES.feedback, models.NAMES.user])
+
+# Name for the exploration search index
+SEARCH_INDEX_EXPLORATIONS = 'explorations'
+
+# Name for the collection search index
+SEARCH_INDEX_COLLECTIONS = 'collections'
+
+#TODO(bhenning): Improve the ranking calculation. Some possible suggestions
+# for a better ranking include using an average of the search ranks of each
+# exploration referenced in the collection and/or demoting collections
+# for any validation errors from explorations referenced in the collection.
+_STATUS_PUBLICIZED_BONUS = 30
+# This is done to prevent the rank hitting 0 too easily. Note that
+# negative ranks are disallowed in the Search API
+_DEFAULT_RANK = 20
+
+def index_exploration_summaries(exp_summaries):
+    """Indexes the explorations corresponding to the given exploration
+    summaries.
+
+    Args:
+        exp_summaries: list(ExpSummaryModel). List of ExplorationSummary
+        domain objects to be indexed.
+    """
+    search_services.add_documents_to_index([_exp_summary_to_search_dict(
```

Comment:
The indentation here is a bit off. Prefer:

    search_services.add_documents_to_index([
        _exp_summary_to_search_dict(exp_summary)
        for exp_summary in exp_summaries
        if _should_index_exploration(exp_summary)
    ], SEARCH_INDEX_EXPLORATIONS)

Ditto below.